docs/adrs/0096-hook-extensions.md

```diff
@@ -0,0 +1,32 @@
+# ADR 0096: Hook extensions
+
+**Date:** 3 August 2023
+
+**Status**: Proposed <!--Accepted|Rejected|Superceded|Deprecated-->
+
+## Context
+
+The current implementation of container hooks does not allow users to customize the pods created by the hook. While the implementation is designed to be used as is or as a starting point, building and maintaining a custom hook implementation just to specify additional fields is not a good user experience.
+
+## Decision
+
+We have decided to add hook extensions to the container hook implementation. This will allow users to customize the pods created by the hook by specifying additional fields. The hook extensions will be implemented in a way that is backwards-compatible with the existing hook implementation.
+
+To allow customization, the runner executing the hook should have `ACTIONS_RUNNER_CONTAINER_HOOK_JOB_TEMPLATE` environment variable pointing to a yaml file on the runner system.
+
+If environment variable is set, but the file can't be read, the hook will fail, signaling incorrect configuration.
+
```

Comment:
If empty or not valid yaml, do we fail?

Reply:
Maybe we should just log it? If someone provides an env, and the file exists, we can issue a warning saying that no fields are used from the file because the file is empty or incorrectly formatted.
Or we can fail here. I don't really know what would be the best approach :relaxed: